packages/contracts/src/2.0.0/protocol/Exchange/MixinSignatureValidator.sol

```diff
@@ -257,4 +262,78 @@ contract MixinSignatureValidator is
         // signature was invalid.)
         revert("SIGNATURE_UNSUPPORTED");
     }
+
+    /// @dev Verifies signature using logic defined by Wallet contract.
+    /// @param hash Any 32 byte hash.
+    /// @param walletAddress Address that should have signed the given hash
+    ///                      and defines its own signature verification method.
+    /// @param signature Proof that the hash has been signed by signer.
+    /// @return True if signature is valid for given wallet..
+    function isValidWalletSignature(
+        bytes32 hash,
+        address walletAddress,
+        bytes signature
+    )
+        internal
+        view
+        returns (bool isValid)
+    {
+        bytes memory calldata = abi.encodeWithSelector(
+            IWallet(walletAddress).isValidSignature.selector,
+            hash,
+            signature
+        );
+        assembly {
+            let cdStart := add(calldata, 32)
+            let success := staticcall(
+                gas,              // forward all gas
+                walletAddress,    // address of Wallet contract
+                cdStart,          // pointer to start of input
+                mload(calldata),  // length of input
+                cdStart,          // write input over output
+                32                // output size is 32 bytes
+            )
+            // Signature is valid if call did not revert and returned true
+            isValid := and(success, mload(cdStart))
```

Comment:
This only works if the called contract returns a non-zero odd number for `true`, like `1`. I'm not sure if the ABI guarantees that boolean true is always encoded as `1` (instead of anything non-zero like in C).

To fix we could do `and(success, iszero(iszero(mload(cdStart))))`.